server/src/main/java/org/elasticsearch/cluster/NodeUsageStatsForThreadPoolsCollector.java

```diff
@@ -47,12 +59,24 @@ public void collectUsageStats(
         ClusterState clusterState,
         ActionListener<Map<String, NodeUsageStatsForThreadPools>> listener
     ) {
-        var dataNodeIds = clusterState.nodes().getDataNodes().values().stream().map(node -> node.getId()).toArray(String[]::new);
+        var dataNodeIds = clusterState.nodes().getDataNodes().values().stream().map(DiscoveryNode::getId).toArray(String[]::new);
+        // Discard last-seen values for any nodes no longer present in the cluster state
+        lastNodeUsageStatsPerNode.keySet().retainAll(Arrays.asList(dataNodeIds));
         if (clusterState.getMinTransportVersion().supports(TRANSPORT_NODE_USAGE_STATS_FOR_THREAD_POOLS_ACTION)) {
             client.execute(
                 TransportNodeUsageStatsForThreadPoolsAction.TYPE,
                 new NodeUsageStatsForThreadPoolsAction.Request(dataNodeIds),
-                listener.map(response -> response.getAllNodeUsageStatsForThreadPools())
+                listener.map(response -> {
+                    // Update last seen stats
+                    lastNodeUsageStatsPerNode.putAll(response.getAllNodeUsageStatsForThreadPools());
+                    if (response.failures().isEmpty() == false) {
+                        logger.warn(
+                            "Got no usage stats from nodes {}",
```

Comment:
I'd like to expand the log message a little bit so it is clearer to people who are not familiar with the PR.
```suggestion
                            "Got no usage stats from nodes [{}], using last known stats for them",
```